### PR TITLE
Parse v6 snapshot to migrate accounts to v7

### PR DESCRIPTION
**Pull request by coin1111**
- [x] parse v6 snapshot from backup tool
- [x]  create legacy recovery structure from snapshot info for each account
- [x] Add support for additonal user structures:
```
Burn::UserBurnPreference
vouch::*
donor_voice_txs::*
fee_maker::FeeMaker
jail::Jail
pledge_account::*
``` 
- [x] add support for framework state:
```
for 0x1 Framework State:
burn::BurnCounter
donor_voice::Registry
fee_maker::EpochFeeMakerRegistry
match_index::MatchIndex;
validator_universe::ValidatorUniverse
slow_wallet::SlowWalletList

``` 

- [ ] Add table state:
```
These are Tables... not sure how to do.
account::OriginatingAddress
account::MigrateOriginatingAddress
```

- [x] add a basic test

- [x] add snapshot digest in tools/storage/json directory

# Necessary For User Migration
- [x] FeeMaker
```
cat state_epoch_79_ver_33217173.795d.json | jq '.[] | select(.fee_maker != null)' | less
```
- [x] Ancestry
```
cat state_epoch_79_ver_33217173.795d.json | jq '.[] | select(.ancestry != null)' | less
```
- [x] CommWallet bool
```
cat state_epoch_79_ver_33217173.795d.json |  jq '[.[] | select(.comm_wallet.list != null) ]' | less
```
- [x] ol_account::BurnTracker

- [x] UserBurnPreference
none
parsed but all null

```
cat state_epoch_79_ver_33217173.795d.json |  jq '[.[] | select(.burn_counter != null) ]' | less
```

# Necessary for system migration
## confirm can be regenerated
- [x] slow registry
- [x] burn::BurnCounter
```
cat state_epoch_79_ver_33217173.795d.json |  jq '[.[] | select(.slow_wallet_list != null) ]' | less
```
- [x] donor voice registry
```
at state_epoch_79_ver_33217173.795d.json |  jq '[.[] | select(.donor_voice_registry != null) ]' | less
```
- [x] EpochFeeMakerRegistry
```
cat state_epoch_79_ver_33217173.795d.json |  jq '[.[] | select(.epoch_fee_maker_registry != null) ]' | less
```

### CODE REVIEW

File: tools/storage/src/read_snapshot.rs [hunk 1]
```diff
@@ -101,8 +135,131 @@ async fn test_deserialize_account() {
...
+    pub slow_wallet_list: Option<SlowWalletListResource>,
+    // TODO: use on V7 tools
+    // ///
+    // pub fullnode_counter: Option<FullnodeCounterResource>,
+    // ///
+    // pub autopay: Option<AutoPayResource>,
+}
+
+
+fn get_legacy_recovery(account_states: &Vec<AccountState>) -> anyhow::Result<Vec<LegacyRecoveryV6>> {
```
Comment (coin1111) [inline tools/storage/src/read_snapshot.rs]: populate legacy recovery here

File: tools/storage/src/read_snapshot.rs [hunk 2]
```diff
@@ -101,8 +135,131 @@ async fn test_deserialize_account() {
...
+            role: AccountRole::EndUser,
+            balance: None,
+            val_cfg: None,
+            miner_state: None,
+            comm_wallet: None,
+            //fullnode_counter: None,
+            //autopay: None,
+            currency_info: None,
+            ancestry: None,
+            make_whole: None,
```
Comment (0o-de-lally) [inline tools/storage/src/read_snapshot.rs]: Make Whole, not necessary

File: tools/storage/src/read_snapshot.rs [hunk 3]
```diff
@@ -101,8 +135,131 @@ async fn test_deserialize_account() {
...
+    // ///
+    // pub fullnode_counter: Option<FullnodeCounterResource>,
+    // ///
+    // pub autopay: Option<AutoPayResource>,
+}
+
+
+fn get_legacy_recovery(account_states: &Vec<AccountState>) -> anyhow::Result<Vec<LegacyRecoveryV6>> {
+    let mut legacy_recovery_v6 = Vec::new();
+    for account_state in account_states.iter() {
```
Comment (0o-de-lally) [inline tools/storage/src/read_snapshot.rs]: Please add for USERS
Burn::UserBurnPreference
vouch::*
donor_voice_txs::*
fee_maker::FeeMaker
jail::Jail
pledge_account::*

for 0x1 Framework State:
burn::BurnCounter
donor_voice::Registry
fee_maker::EpochFeeMakerRegistry
match_index::MatchIndex;
validator_universe::ValidatorUniverse
slow_wallet::SlowWalletList

These are Tables... not sure how to do.
account::OriginatingAddress
account::MigrateOriginatingAddress

Comment (coin1111) [reply tools/storage/src/read_snapshot.rs]: so far added support for 
```
Burn::UserBurnPreference
vouch::*
donor_voice_txs::*
fee_maker::FeeMaker
jail::Jail
pledge_account::*
```

**Pull request merged**